### PR TITLE
Update "TI Map" analytics rules for performance

### PR DESCRIPTION
**Pull request by kfriede**
Required items, please complete
   
   Change(s):
   - Changed query to put any applicable WHERE filters before any SUMMARIZE calls for performance. 
   - Added ThreatIntelligenceTaxii data connector on missing analytics.

   Reason for Change(s):
   - Many of these queries would fail on real-world datasets due to summarization of unnecessary data. 
   - ThreatIntelligenceTaxii is a valid provider of ThreatIntelligenceIndicator table data and should be included

   Version Updated:
   - Yes

   Testing Completed:
   - Yes, as much as possible (don't have all data sources to test appropriately for some rules)

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

**Comment by kfriede** [comment]
To whoever ends up picking this up, the KQL validation failures is due to Line 397 of  https://github.com/Azure/Azure-Sentinel/blob/master/.script/tests/KqlvalidationsTests/KqlValidationTests.cs 

The regex being used is overly-specific and not conducive for the performance improvements being made here.

**Comment by v-rbajaj** [comment]
Hi @kfriede, thanks for raising this PR. We will review this PR by 13 Oct 2023.

**Comment by kfriede** [comment]
This traces back to commit [16564be](https://github.com/Azure/Azure-Sentinel/commit/16564be03e3302e33a19e035a36e0b84c8426d44) from @aprakash13

**Comment by v-rbajaj** [comment]
Hi @kfriede, for KQL validation, we will see what can be done in this case. 

We would need you to repackage this solution using this tool https://github.com/Azure/Azure-Sentinel/tree/master/Tools/Create-Azure-Sentinel-Solution/V3

Thanks

**Comment by v-rbajaj** [comment]
Hi @kfriede, please provide update on above comment.

**Comment by v-rbajaj** [comment]
Hi @kfriede, please provide some update on this PR

**Comment by kfriede** [comment]
Hi @v-rbajaj, apologies I was out last week. I will look at repackaging the solution for these edits

**Comment by v-rbajaj** [comment]
> Hi @v-rbajaj, apologies I was out last week. I will look at repackaging the solution for these edits
> 
> Edit: Solution has been repackaged for 3.0.2

Thanks @kfriede, will review the PR by 27 Oct, 2023.

File: Solutions/Threat Intelligence/Package/createUiDefinition.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
...
+        "description": "<img src=\"https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/Logos/Azure_Sentinel.svg\"width=\"75px\" height=\"75px\">\n\n**Note:** _There may be [known issues](https://aka.ms/sentinelsolutionsknownissues) pertaining to this Solution, please refer to them before installing._\n\nThe Threat Intelligence solution contains data connectors for import of threat indicators into Microsoft Sentinel, analytic rules for matching TI data with event data, workbook, and hunting queries. Threat indicators can be malicious IP's, URL's, filehashes, domains, email addresses etc.\n\n**Workbooks:** 1, **Analytic Rules:** 38, **Hunting Queries:** 5\n\n[Learn more about Microsoft Sentinel](https://aka.ms/azuresentinel) | [Learn more about Solutions](https://aka.ms/azuresentinelsolutionsdoc)",
```
Comment (v-rbajaj) [inline Solutions/Threat Intelligence/Package/createUiDefinition.json:9]: Hi @kfriede, Please revert this change.

Comment (kfriede) [reply Solutions/Threat Intelligence/Package/createUiDefinition.json]: @v-rbajaj done!

**Comment by v-rbajaj** [comment]
Hi @kfriede, thanks. We are currently looking into the KQL validations and once that is fixed, we will revisit this PR by 2 Nov 2023.

**Comment by v-rbajaj** [comment]
Hi @kfriede, we are still investigating on KQL validations, will get back to by 03 Nov 2023.

**Comment by kfriede** [comment]
Hi @v-rbajaj, it's been over 1 week since your last update. Is this still under review?

**Comment by v-atulyadav** [comment]
Hi @kfriede,
Apologies for the inconvenience, but due to unavailability, the review has been delayed. We assure you that you will receive an update on this by the end of the day tomorrow. Thank you.

**Comment by v-rbajaj** [comment]
Hi @kfriede, we are working on the KQL changes and we will soon provide an update.

File: Solutions/Threat Intelligence/Analytic Rules/DomainEntity_CommonSecurityLog.yaml [hunk 2]
```diff
@@ -43,10 +43,10 @@ query: |
   let Domain_Indicators =
       ThreatIntelligenceIndicator
       | where TimeGenerated >= ago(ioc_lookBack)
-      | summarize LatestIndicatorTime = arg_max(TimeGenerated, *) by IndicatorId
       | where Active == true and ExpirationDateTime > now()
```
Comment (blauwers) [inline Solutions/Threat Intelligence/Analytic Rules/DomainEntity_CommonSecurityLog.yaml:46]: This where statement needs to follow the summarize statement. Same for the other KQL queries.

**Review by blauwers — CHANGES_REQUESTED** [review]

**Comment by kfriede** [comment]
@v-rbajaj after refactoring, files now give: `Kqlvalidations.Tests.KqlValidationTests.Validate_DetectionQueries_SkippedTemplatesDoNotHaveValidKql(fileName: "IPEntity_DuoSecurity.yaml", encodedFilePath: "L2hvbWUvdnN0cy93b3JrLzEvcy9Tb2x1dGlvbnMvVGhyZWF0IE"...)` in testing

**Comment by v-rbajaj** [comment]
Hi @kfriede, sorry for the above commits, was trying to fix few errors from KQL validations. We are checking on it, will get back to you by 29 Nov 2023.

**Comment by kfriede** [comment]
@v-rbajaj I’m good with #9369, and we can close this (#9181) when #9369 is merged


--
Kevin Friedemann

**Comment by v-rbajaj** [comment]
Thanks @kfriede, I have changed the PR status from open to draft and will close later once #9369 is merged.

**Comment by v-rbajaj** [comment]
Closing this PR as #9369 is merged.

**Pull request closed without merging**